src/main/java/seedu/address/model/patient/Patient.java

```diff
@@ -62,10 +62,28 @@ public Patient(Name name, Phone phone, Email email, Gender gender, IcNumber icNu
         this.birthday = birthday;
         this.address = address;
         this.tags.addAll(tags);
-        this.assignedDepartment = assignedDepartment; // default Department given
+        this.assignedDepartment = assignedDepartment;
         this.record = new Record(this); // creates new Record
     }
 
+    /**
+     * Patient constructor with all fields as parameters
+     */
+    public Patient(Name name, Phone phone, Email email, Gender gender, IcNumber icNumber, Birthday birthday,
+                   Address address, Set<Tag> tags, AssignedDepartment assignedDepartment, Record record) {
+        requireAllNonNull(name, phone, email, icNumber, birthday, address, tags);
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.gender = gender;
+        this.icNumber = icNumber;
+        this.birthday = birthday;
+        this.address = address;
+        this.tags.addAll(tags);
+        this.assignedDepartment = assignedDepartment;
+        this.record = record;
```

Comment:
Hmm, perhaps it will be great to update the `Record` class as well, to make a copy of the old Record information but update its patient to be the new Patient object? Because right now the record object parameter contains the old record with its patient field being the old Patient object, and it might lead to problems whereby there is a mismatch between the new Patient object storing this Record object, and the Record object storing the old Patient object as its patient (an immutable field). --> Create a `copyRecordInformation` method or something similar maybe?

Reply:
Thanks for your suggestion! I've made the relevant changes.